schedule-replicas-by-topology-labels.md

```diff
@@ -0,0 +1,131 @@
+
+Take the topology in the previous section as an example.
+
+Assume that the number of cluster replicas is 3 (`max-replicas=3`). As there are 3 zones in total, PD keeps the 3 replicas of each Region placed in z1/z2/z3, so that the TiDB cluster is still available when any data center fails,
+
+Assume that the number of cluster replicas is 5 (`max-replicas=5`). As there are only 3 zones in total, PD can not guarantee the isolation of each replica at this level. At this time, the PD scheduler adjusts to ensure isolation at the host level. In other words, multiple copies of a Region might be distributed in the same zone, but will not be distributed on the same host.
+
+Under the premise of the 5-replica configuration, if z3 occurs an overall failure or isolation, and can not be restored in a period (controlled by `max-store-down-time`), PD will make up the 5 replicas through scheduling. At this time, only 3 hosts are available, so host level isolation can not be guaranteed. Therefore, multiple copies might be distributed to the same host. But if the value `isolation-level` is set to `zone` instead of empty, this specifies the minimum physical isolation requirements for Region replicas. That is to say, PD ensures that replicas of the same Region are scattered in different zones. Even if following this isolation restriction does not meet the multiple replica requirements of `max-replicas`, PD will not perform corresponding scheduling. 
+
+For example, there are three data centers z1, z2 and z3 in the TiKV cluster. Under the three-replica setting, PD distributes the three replicas of the same Region to these three data centers respectively. At this time, if the entire data center z1 encounters a power outage and cannot be recovered in a period, the PD will consider that the Region copy on z1 is no longer available. But because the `isolation-level` is set to `zone`, PD needs to strictly ensure that different Region replicas will not fall into the same zone. At this time, as both z2 and z3 already have replicas, PD does not perform any scheduling under the restriction of the minimum mandatory isolation level, even if there are only two replicas.
+
+Similarly, when `isolation-level` is set to `rack`, the minimum isolation level is different racks in the same data center. Under this setting, if isolation can be guaranteed at the zone level, it will be guaranteed first. Only when the zone level isolation can not be completed, will the scheduling in the same zone and the same rack be considered to avoid.
+
+In summary, PD maximizes the disaster recovery of the cluster according to the current topology. Therefore, if you want to achieve a certain level of disaster recovery, deploy more machines on different sites according to the topology than the number of `max-replicas`. Also, TiDB provides mandatory isolation level settings such as `isolation-level` to control the topological isolation level according to the scenario more flexibly.
```

Comment:
```suggestion
In summary, PD maximizes the disaster recovery of the cluster according to the current topology. Therefore, if you want to achieve a certain level of disaster recovery, deploy more machines on different sites according to the topology than the number of `max-replicas`. TiDB also provides a mandatory configuration item `isolation-level` to control the topological isolation level of data according to different cases.
```

Reply:
```suggestion
In summary, PD maximizes the disaster recovery of the cluster according to the current topology. Therefore, if you want to achieve a certain level of disaster recovery, deploy more machines on different sites according to the topology than the number of `max-replicas`. TiDB also provides mandatory configuration items such as `isolation-level` to control the topological isolation level of data according to different cases.
```
I've updated my version according to the source text:
> 同时 TiDB 也提供了诸如 isolation-level 这样的强制隔离级别设置，以便更灵活地根据场景来控制对数据的拓扑隔离级别。

Reply:
To me, `mandatory isolation level settings ` is not clear enough. It's about language clarity. For other parts, I've updated my comment, PTAL.